File: webscraping/youtube.com/YouTubeSearch.java
// https://kaijento.github.io/2017/05/19/web-scraping-youtube.com/
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

import java.io.IOException;

public class YouTubeSearch {

    public static void main (String[] args) throws IOException {

        String url   = "http://www.youtube.com/results";
        String query = "dj liquid raving";

        Document doc = Jsoup.connect(url)
            .data("search_query", query)
            .userAgent("Mozilla/5.0")
            .get();

        for (Element a : doc.select(".yt-lockup-title > a[title]:not([href*=&list=])")) {
            System.out.println(
                "http://www.youtube.com" + a.attr("href") + " " + a.attr("title")
            );
        }

    }
}

File: webscraping/youtube.com/youtube-search.py
# https://kaijento.github.io/2017/05/19/web-scraping-youtube.com/
import requests
from   bs4 import BeautifulSoup

with requests.session() as s:
    s.headers['user-agent'] = 'Mozilla/5.0'

    url    = 'http://www.youtube.com/results'
    params = {'search_query': 'dj liquid raving'}

    r    = s.get(url, params=params)
    soup = BeautifulSoup(r.content, 'html5lib')
    
    for a in soup.select('.yt-lockup-title > a[title]'):
        if '&list=' not in a['href']:
            print('http://www.youtube.com' + a['href'], a['title'])

